uint8[2] head
uint8 length
ProtocolInner inner_msg
uint16 crc